# Avaliação Estatística de um banco de dados

Universidade Federal Rural de Pernambuco

Bacharelado em Ciência da Computação

Estatística Exploratória - Prof. Cláudio Cristino

Segunda Verificação de Aprendizagem - em 26/05/2022

Steffano Xavier Pereira

## Base de dados

Inicialmente vamos carregar o nosso banco de dados, para que possamos trabalhar em nosso ambiente virtual Python e fazer análises de acordo com o que foi proposto para a atividade.

In [2]:
import pandas as pd

# Importando os dados
dados = pd.read_csv('./data/Steffano Pereira - Dados_individuais_2VA_Est_Exploratoria - Steffano.csv', index_col='ID')

Podemos também visualizar um corte dos 5 primeiros dados, para confirmar que os dados estão organizados da maneira que desejamos:

In [3]:
dados.head()

Tipo       Categoria  Download (número)  Avaliação (0 - 5)  \
ID                                                                     
App1      Pago  Entreterimento              94506                  4   
App2  Gratuito      Utilitário             281962                  4   
App3      Pago  Entreterimento             119282                  3   
App4  Gratuito  Entreterimento             136822                  3   
App5  Gratuito     Rede Social             286821                  1   

      Tamanho (Mb)  Tempo no ar (em dias)  
ID                                         
App1            63                    349  
App2           120                    365  
App3            70                    335  
App4            74                    331  
App5           123                    282

Como obtemos os valores em nossa variável `dados`, podemos continuar para as nossas atividades.

## Amostras

Para essa atividade também é necessário construir amostras que irão servir como referência para o estudo de nossa população. Para tal, o conjunto das amostras serão demonstrados nessa seção.

Vale observar que as amostras são do tipo de amostra aleatória simples.

### Amostras sobre o Tamanho (Mb)

Vamos selecionar algumas amostras sob o tamanho dos aplicativos encima de nossa população.

In [102]:
amostra_tam10 = dados['Tamanho (Mb)'].sample(n=10).to_frame()
amostra_tam50 = dados['Tamanho (Mb)'].sample(n=50).to_frame()
amostra_tam100 = dados['Tamanho (Mb)'].sample(n=100).to_frame()

amostra_tam = [amostra_tam10, amostra_tam50, amostra_tam100]

## 1. Análise Descritiva

Devemos fazer uma análise descritiva de todas as variáveis, para isso, vamos estabelecer os seguintes critérios:

- Devemos analisar as variáveis qualitativas separadamente
- É interessante observar análises de aplicativos de categorias diferentes

### 1.1 Análise Geral

Inicialmente, podemos analisar dados descritivos de **toda nossa amostra** com a função `describe()`.

Considere nesse quesito que:

- count - Número de exemplos
- mean - Média
- std - Desvio Padrão
- min - Valor mínimo
- 25% - Primeiro quartil
- 50% - Segundo Quartil/Mediana
- 75% - Terceiro Quartil 
- max - Valor Máximo

In [103]:
dados.describe()

Download (número)  Avaliação (0 - 5)  Tamanho (Mb)  \
count         316.000000         316.000000    316.000000   
mean       167952.481013           3.598101     82.810127   
std         87128.578413           1.017246     30.090267   
min         12954.000000           1.000000     10.000000   
25%         94472.250000           3.000000     63.000000   
50%        175262.000000           4.000000     84.000000   
75%        242500.750000           4.000000    103.000000   
max        309906.000000           5.000000    183.000000   

       Tempo no ar (em dias)  
count             316.000000  
mean              353.224684  
std                36.187916  
min               258.000000  
25%               329.000000  
50%               354.000000  
75%               376.250000  
max               467.000000

Para os atributos quantitativos, podemos tirar algumas conclusões:

- Download

    Podemos observar o número de Downloads como um intervalo entre `12.954` à `309.906`, do qual podemos concluir que nossa amostra possui aplicativos na ordem das dezena de milhares até 3 centena de milhares, do qual pode se considerar que temos aplicativos com um nível de fama alto, logo quanto maior o número de downloads, maior é o público-alvo atingido por esse aplicativo.
    
- Avaliação

    Na avaliação, podemos observar um fato interessante, que é a média e seu desvio padrão; em geral, a avaliação parece balanceada, pois provavelmente, devemos ter aplicativos com boas avaliações e também aplicativos más avaliações, com o primeiro quartil (25%) até avaliação `3` e do segundo (50%) e terceiro quartil (75%) avaliações maiores que `4`, daqui já podemos concluir que há um grupo marginal/pequeno de avaliações `1` e `2`; e um grupo considerável de avaliações entre `4` e `5`. Vale observar que nessa amostra não há avaliações com valor `0`.

- Tamanho

    No tamanho, é interessante observar que temos um intervalo bem definido entre `10` mb até `183` mb. No geral, os aplicativos da amostra possuem uma média de `82,81` mb e desvio padrão de `30,09` mb. De modo geral, estão bem distribuídos de acordo com seus quartis, principalmente que há uma minoria menor que `63` mb. O tamanho será importante de observar, quando estudarmos as categorias em específico.
    
- Tempo no ar

    Já para o tempo no ar, um ponto muito importante de se observar, é que o mínimo de dias equivale há `258` dias, o que corresponde a um período maduro de dias maior que 6 meses de lançamento, o que torna a amostra madura o suficiente para o estudo.

Podemos separar nossa amostra por tipo e categoria e assim analisar cada uma individualmente.

### 1.2 Análise Descritiva por Tipo

Para analisar os aplicativos pagos de aplicativos gratuitos é necessário separar os dados de acordo, e em seguida observar suas relações descritivas.

In [104]:
apps_pagos = dados[dados.Tipo == 'Pago']
apps_gratuitos = dados[dados.Tipo == 'Gratuito']

#### Aplicativos Pagos

In [105]:
apps_pagos.describe()

Download (número)  Avaliação (0 - 5)  Tamanho (Mb)  \
count         116.000000         116.000000    116.000000   
mean        70553.181034           3.491379     51.637931   
std         34263.839918           0.991228     15.717283   
min         12954.000000           1.000000     10.000000   
25%         40461.750000           3.000000     41.750000   
50%         75427.000000           4.000000     56.500000   
75%         98713.000000           4.000000     64.000000   
max        128532.000000           5.000000     72.000000   

       Tempo no ar (em dias)  
count             116.000000  
mean              349.077586  
std                35.145450  
min               258.000000  
25%               326.500000  
50%               350.500000  
75%               368.250000  
max               446.000000

#### Aplicativos Gratuitos

In [106]:
apps_gratuitos.describe()

Download (número)  Avaliação (0 - 5)  Tamanho (Mb)  \
count         200.000000         200.000000    200.000000   
mean       224444.075000           3.660000    100.890000   
std         50998.471599           1.029417     19.887318   
min        130861.000000           1.000000     73.000000   
25%        180537.250000           3.000000     85.000000   
50%        224285.500000           4.000000     97.000000   
75%        269094.250000           4.000000    113.000000   
max        309906.000000           5.000000    183.000000   

       Tempo no ar (em dias)  
count             200.000000  
mean              355.630000  
std                36.650778  
min               265.000000  
25%               330.000000  
50%               356.000000  
75%               383.250000  
max               467.000000

### 1.2 Análise Descritiva por Categoria

In [107]:
apps_utilitario = dados[dados.Categoria == 'Utilitário']
apps_entretenimento = dados[dados.Categoria == 'Entreterimento']
apps_redesSocial = dados[dados.Categoria == 'Rede Social']
apps_cientifico = dados[dados.Categoria == 'Científico']


In [108]:
apps_utilitario.describe()

Download (número)  Avaliação (0 - 5)  Tamanho (Mb)  \
count           94.00000          94.000000     94.000000   
mean        193106.12766           4.351064     90.297872   
std          83619.51466           0.479862     31.111603   
min          12954.00000           4.000000     10.000000   
25%         146756.50000           4.000000     76.250000   
50%         206488.50000           4.000000     92.000000   
75%         256595.50000           5.000000    108.000000   
max         308516.00000           5.000000    157.000000   

       Tempo no ar (em dias)  
count              94.000000  
mean              378.989362  
std                12.589033  
min               360.000000  
25%               368.000000  
50%               375.500000  
75%               389.750000  
max               405.000000

In [109]:
apps_entretenimento.describe()

Download (número)  Avaliação (0 - 5)  Tamanho (Mb)  \
count         138.000000         138.000000    138.000000   
mean       153102.144928           3.282609     78.398551   
std         89825.282518           0.566577     30.941324   
min         14886.000000           2.000000     16.000000   
25%         79786.750000           3.000000     58.250000   
50%        129696.500000           3.000000     72.500000   
75%        234236.000000           4.000000     99.750000   
max        309906.000000           4.000000    183.000000   

       Tempo no ar (em dias)  
count             138.000000  
mean              340.362319  
std                14.265953  
min               303.000000  
25%               331.000000  
50%               343.000000  
75%               350.750000  
max               364.000000

In [110]:
apps_redesSocial.describe()

Download (número)  Avaliação (0 - 5)  Tamanho (Mb)  \
count          49.000000          49.000000     49.000000   
mean       183315.510204           2.122449     87.122449   
std         69331.964371           0.725507     21.148909   
min         27824.000000           1.000000     33.000000   
25%        137477.000000           2.000000     74.000000   
50%        193074.000000           2.000000     89.000000   
75%        239272.000000           3.000000    102.000000   
max        293256.000000           3.000000    128.000000   

       Tempo no ar (em dias)  
count              49.000000  
mean              299.469388  
std                18.210691  
min               258.000000  
25%               289.000000  
50%               302.000000  
75%               316.000000  
max               322.000000

In [111]:
apps_cientifico.describe()

Download (número)  Avaliação (0 - 5)  Tamanho (Mb)  \
count          35.000000          35.000000     35.000000   
mean       137441.485714           4.885714     74.057143   
std         89249.778169           0.322803     30.202204   
min         26949.000000           4.000000     32.000000   
25%         55620.000000           5.000000     49.000000   
50%        114270.000000           5.000000     68.000000   
75%        209662.000000           5.000000     92.500000   
max        305825.000000           5.000000    146.000000   

       Tempo no ar (em dias)  
count              35.000000  
mean              410.000000  
std                20.531181  
min               380.000000  
25%               393.500000  
50%               410.000000  
75%               414.500000  
max               467.000000

Observando as categorias, podemos observar que há uma categoria dominante, que é a de entretenimento, com `138` aplicativos. Observando o **número de Downloads**, temos a média dos mais baixados são os **aplicativos utilitários** e também com maior **tamanho**. Os **aplicativos científicos** compõem um grupo que possuem **melhor avaliação** e também com maior **tempo no ar**, porém vale notar que, os **aplicativos científicos** são minoria na amostra por categoria e representam apenas `35` dos `316`, esse valor por sua vez pode representar apenas uma parcela pequena se comparada com toda a população de aplicativos científicos; logo, vale ter cuidado ao tirar conclusões e outros estudos são necessários para uma maior certeza.

## 2. Estimação Pontual

Para a Estimação Pontual, vamos utilizar o conjunto de amostras de Tamanho, veja que essa amostragem é simples e aleatória, e com a estimação Pontual buscamos observar os valores da amostra para estimar os valores da nossa população. Iremos trabalhar com 3 amostras (10, 50 e 70 exemplos de um total de 316 exemplos da população) que irão tentar estimar o valor da população.

### 2.1 Seleção dos estimadores

Vamos utilizar 2 estimadores para esse conjunto de dados, a **média amostral** e também a **variância amostral**. Pois a média amostral considera todos os valores de nossa amostra e a variância garante a exatidão do nosso conjunto de dados em torno de nossa média.

### 2.2 Criando nossos estimadores amostrais

Para isso vamos utilizar duas ferramentas da biblioteca `pandas`: `mean()` que consiste na função que obtém a média de um conjunto de dados e `var()` que consiste na função que obtém a variância não viciada de um conjunto de dados.

In [112]:
# Função para calcular a estimativa de uma amostra
def calcular_estimacao_amostral(amostra: pd.DataFrame):
    media_amostral = float(amostra.mean())
    variancia_amostral = float(amostra.var())

    return media_amostral, variancia_amostral

### 2.3 Calculando as Estimativas

Vamos estimar por meio do cálculo amostral de média e variância e iremos comparar o resultado com os parâmetros.

In [113]:
estimativa = []
for amostra in amostra_tam:
    estimativa.append(calcular_estimacao_amostral(amostra))

Vamos comparar as estimativas do `Tamanho (Mb)` com o valor do parâmetro.

In [118]:
dados_tam = dados['Tamanho (Mb)']

print("Estudo sobre a população de aplicativos e seus respectivos tamanhos e")
print("amostras diversas que aproximam suas estimativas com os parâmetros.\n")
print(f"Parâmetro: ({dados_tam.mean()}, {dados_tam.var()})")
print(f"Amostra 1: {estimativa[0]}")
print(f"Amostra 2: {estimativa[1]}")
print(f"Amostra 3: {estimativa[2]}")


Estudo sobre a população de aplicativos e seus respectivos tamanhos e
amostras diversas que aproximam suas estimativas com os parâmetros.

Parâmetro: (82.81012658227849, 905.4241510950372)
Amostra 1: (103.3, 642.0111111111112)
Amostra 2: (79.84, 910.3004081632653)
Amostra 3: (80.27, 882.5627272727273)


Como visto na seção de amostras, separamos em 3 amostras aleatórias simples:

- Amostra 1: 10 exemplos
- Amostra 2: 50 exemplos
- Amostra 3: 100 exemplos

Observe que a amostra 1 se distancia bastante do parâmetro, todavia quando vemos a amostra 2 e 3, podemos observar que essas duas se aproximam ainda mais do nosso parâmetro. Dessa forma vale observar que quanto maior o número de exemplos a estimativa é mais precisa do nosso parâmetro.

## 3) Análise Intervalar

## 4) Teste de Hipótese para a média

## 5) Teste de Hipótese de Aderência ou Independência ou Homogeneidade

## 6) Estudo de Regressão